conda/notices/types.py

```diff
@@ -76,8 +77,15 @@ def _parse_iso_timestamp(iso_timestamp: Optional[str]) -> Optional[datetime]:
 
     @classmethod
     def get_cache_key(cls, url: str, name: str, cache_dir: Path) -> Path:
-        """Returns the place where this channel response will be stored as cache"""
+        """
+        Returns the place where this channel response will be stored as cache
+
+        We use a naive hashing algorithm to avoid issues with filenames (e.g. illegal or
+        problematic characters). We don't care the that this is a weak hash.
+        """
         url_obj = parse.urlparse(url)
-        path = url_obj.path.replace("/", "-")
-        cache_filename = f"{name}{path}"
+        bytes_filename = f"{name}{url_obj.path}".encode()
```

Comment:
bold move calling `.encode()` with no arguments, usually this is spelled `.encode('utf-8')`

Reply:
Meh, I dislike reiterating the [default value](https://docs.python.org/3/library/stdtypes.html#str.encode) 🤷🏻‍♂️

Reply:
Cool, as long as it's not `sys.getdefaultencoding()`, I see that the default is always "utf-8" at least since Python 3.5+

Reply:
I'm curious about the channel name / path instead of the full URL. Conda usually expects channel names to be unique right?